tests/s25Main/UI/testSmartBitmap.cpp

```diff
@@ -278,4 +352,61 @@ BOOST_AUTO_TEST_CASE(MultiPlayerBitmap)
     }
 }
 
+BOOST_AUTO_TEST_CASE(DrawPercent)
+{
+    // Make sure that test passes across different image heights, including some random ones
+    auto sizes =
+      helpers::make_array<Extent>(Extent{rttr::test::randomValue<unsigned int>(10, 50), 50},
+                                  Extent{rttr::test::randomValue<unsigned int>(10, 50), 35},
+                                  rttr::test::randomPoint<Extent>(10, 50), rttr::test::randomPoint<Extent>(10, 50),
+                                  rttr::test::randomPoint<Extent>(10, 50), rttr::test::randomPoint<Extent>(10, 50));
+
+    RTTR_STUB_FUNCTION(glVertexPointer, rttrOglMock2::glVertexPointer);
+    RTTR_STUB_FUNCTION(glDrawArrays, rttrOglMock2::glDrawArrays);
+    RTTR_STUB_FUNCTION(glEnableClientState, rttrOglMock2::glEnableClientState);
+    RTTR_STUB_FUNCTION(glTexCoordPointer, rttrOglMock2::glTexCoordPointer);
+    RTTR_STUB_FUNCTION(glColorPointer, rttrOglMock2::glColorPointer);
+    RTTR_STUB_FUNCTION(glDisableClientState, rttrOglMock2::glDisableClientState);
+    RTTR_STUB_FUNCTION(glBindTexture, rttrOglMock2::glBindTexture);
+
+    for(const auto& size : sizes)
+        BOOST_TEST_CONTEXT("Bitmap size: " << size)
+        {
+            auto bmpSrc = createBmp(size, 0);
+            glSmartBitmap smartBmp;
+            smartBmp.add(bmpSrc.get());
+
+            // drawPercent(0) shouldn't do a thing
+            rttrOglMock2::callCount = 0;
+            smartBmp.drawPercent(DrawPoint::all(0), 0);
+            BOOST_TEST(rttrOglMock2::callCount == 0);
+
+            auto ratio = rttr::test::randomValue<float>(.1f, .9f);
+            BOOST_TEST_CONTEXT("Draw percentage: " << ratio * 100)
+            {
+                smartBmp.drawPercent(DrawPoint::all(0), helpers::iround<int>(100 * ratio));
+                BOOST_TEST(rttrOglMock2::textureCoords.size() == size_t(4));
+                // All top texture coords have same Y component
+                BOOST_TEST(rttrOglMock2::textureCoords[0].y == rttrOglMock2::textureCoords[3].y);
+                // All bottom texture coords have same Y component
+                BOOST_TEST(rttrOglMock2::textureCoords[1].y == rttrOglMock2::textureCoords[2].y);
+                // Bottom starts at bottom
+                BOOST_TEST(rttrOglMock2::textureCoords[1].y == smartBmp.texCoords[1].y);
+                // Top starts at given percentage, draw accuracy is +- 1px
+                auto expectedPixelHeight = ratio * size.y;
+                auto actualRatio =
+                  ((rttrOglMock2::textureCoords[1].y - rttrOglMock2::textureCoords[0].y) / smartBmp.texCoords[1].y);
```

Comment:
Nit: Take the actual start coordinate into account:
```suggestion
                const auto actualRatio =
                  (rttrOglMock2::textureCoords[1].y - rttrOglMock2::textureCoords[0].y) / (smartBmp.texCoords[1].y -  smartBmp.texCoords[0].y);
```